### ISSUE TITLE
CA3147 for custom implementations of ValidateForgeryTokenAttributes

### ISSUE THREAD

**Issue by PBuchmann** [dotnet/roslyn-analyzers#2175]
#### Analyzer package

[Microsoft.NetFramework.Analysers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.6.3

#### Diagnostic ID

Example: [CA3147]

#### Repro steps

I've implemented a custom forgery token validation that reads the token from a cookie. This was a good fit for me for JSON requests. Here's my code.

    `/// <summary>
    /// Attribut zum Filtern von RequestValidationTokens im Header.
    /// </summary>
    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
    public class ValidateJsonAntiForgeryTokenAttribute : FilterAttribute, IAuthorizationFilter
    {
        /// <inheritdoc/>
        public void OnAuthorization(AuthorizationContext filterContext)
        {
            if (filterContext == null)
            {
                throw new ArgumentNullException(nameof(filterContext));
            }

            var httpContext = filterContext.HttpContext;
            var cookie = httpContext.Request.Cookies[AntiForgeryConfig.CookieName];
            AntiForgery.Validate(cookie != null ? cookie.Value : null, httpContext.Request.Headers[AntiForgeryConfig.CookieName]);
        }
    }`

CA3147 only validates the default attribute ValidateAntiForgeryTokenAttribute. Would there be a chance to fulfill the rule also with such custom anti forgery token attributes? 

#### Expected behavior

Rule CA3147 also accepts custom ValidateAntiForgeryTokenAttribute like above and so does not have to be suppressed in such wrong positive cases.

#### Actual behavior

MVC actions that are attributed with [ValidateJsonAntiForgeryToken] are also accepted to be OK.

**Comment by rsullivan00** [comment]
@PBuchmann did you find a workaround for this case?

We're running into the same issue currently--we'd like to use a custom CSRF-verification implementation, but also want code analysis to verify that CSRF-protection is added to new endpoints.

**Comment by PBuchmann** [comment]
No, I had given up on this. But honestly I wouldn't call me a major expert on this topic. I just stumbled over it.
If I remember correclty, I was forced to suppress the messages on each occurance. 
Maybe the problem can be that a custom forgery attribute could pretend security to be existent where it actually isn't. So the analyzer would have to validate the security of the custom attribute. But that's just a guess.
